src/analysisd/analysisd.c

```diff
@@ -906,7 +906,8 @@ void OS_ReadMSG_analysisd(int m_queue)
         os_calloc(1, sizeof(Eventinfo), lf);
         os_calloc(Config.decoder_order_size, sizeof(DynamicField), lf->fields);
         lf->year = prev_year;
-        strncpy(lf->mon, prev_month, 3);
+        strncpy(lf->mon, prev_month, 4);
+        lf->mon[3] = '\0';
```

Comment:
Just my two cents: if we do this, we need to make sure prev_month is always zero-terminated.

Reply:
Hi @RamiroRD 

The variable was initialized with all elements to 0 in the main function before calling this line code. Also, the prev_month is a global variable so, it will initialize to 0 at compile-time and it is initialization is always with a string of the same size and NULL termination. 

Best regards
Hanes